universal-application-tool-0.0.1/app/modules/SecurityModule.java

```diff
@@ -103,19 +109,63 @@ protected ProfileFactory provideProfileFactory(
     return new ProfileFactory(clock, dbContext, httpContext);
   }
 
+  @Provides
+  @Singleton
   protected FakeAdminClient fakeAdminClient(ProfileFactory profileFactory) {
     return new FakeAdminClient(profileFactory);
   }
 
+  @Named("idcs")
```

Comment:
question: my team usually uses [BindingAnnotations](https://g3doc.corp.google.com/third_party/java_src/guice/g3doc/BindingAnnotations.md?cl=head) instead of `@Named` and I'm not sure which is preferred - @bion thoughts?

With binding annotations, you end up with `@` annotations for instances, which I think looks nice. For example, this could be passed to a constructor as `MyClass(@Idcs OidcClient idcsClient)`

Has less chance of misspelling the Strings passed to `@Named`, since the annotations are checked at compile time

Reply:
I commented on the same thing but this is a much higher quality comment!